app/controllers/site_controller.rb

```diff
@@ -1410,24 +1410,37 @@ def set_selected_annotation
     # determine which URL param to use for selection and construct base object
     selector = params[:annotation].nil? ? params[:gene_set_annotation] : params[:annotation]
     annot_name, annot_type, annot_scope = selector.nil? ? @study.default_annotation.split('--') : selector.split('--')
+
     # construct object based on name, type & scope
```

Comment:
Is there a way to refactor this so that this calls  the method in ApiParamsUtils?  Or is the logic different.
Also, maybe we need a better name for ApiParamsUtils since the utility methods may not be restricted to API requests.

Reply:
Possibly - let me look it over.  I agree that the duplication isn't ideal, but the difference in the controllers made me think that it wasn't a good idea.  But I'll take a closer look.

Reply:
Don't know why I thought that wouldn't be a good idea - much cleaner now.  0086954.